src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BinderTypeModelBinder.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// An <see cref="IModelBinder"/> for models which specify an <see cref="IModelBinder"/> using
+    /// <see cref="BindingInfo.BinderType"/>.
+    /// </summary>
+    public class BinderTypeModelBinder : IModelBinder
+    {
+        private readonly ObjectFactory _factory;
+
+        /// <summary>
+        /// Creates a new <see cref="BinderTypeModelBinder"/>.
+        /// </summary>
+        /// <param name="binderType">The <see cref="Type"/> of the <see cref="IModelBinder"/>.</param>
+        public BinderTypeModelBinder(Type binderType)
+        {
+            if (binderType == null)
+            {
+                throw new ArgumentNullException(nameof(binderType));
+            }
+
+            if (!typeof(IModelBinder).GetTypeInfo().IsAssignableFrom(binderType.GetTypeInfo()))
+            {
+                throw new InvalidOperationException(
```

Comment:
At this level, should be an `ArgumentException`. But why doesn't the `BinderTypeModelBinderProvider` do this check? (`InvalidOperationException` would make sense there.)